### PR TITLE
New Lint quick-fix: await_only_futures

### PR DESCRIPTION
**Pull request by JPaulsen**
From [Linter](http://dart-lang.github.io/linter/lints/await_only_futures.html)

### CODE REVIEW

File: pkg/analysis_server/lib/src/services/correction/fix_internal.dart [hunk 1]
```diff
@@ -1784,6 +1787,18 @@ class FixProcessor {
     _addFix(DartFixKind.ADD_NE_NULL, []);
   }
 
+  void _addFix_removeAwait() {
+    AwaitExpression awaitExpression = node is AwaitExpression
+        ? node
+        : node.getAncestor((node) => node is AwaitExpression);
```
Comment (bwilkerson) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: Will the `node` every *not* be an await expression? Just wondering whether this defensive code could be removed. (Here and in other fix methods.)

Comment (JPaulsen) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: Fixed

File: pkg/analysis_server/lib/src/services/correction/fix_internal.dart [hunk 2]
```diff
@@ -1784,6 +1787,16 @@ class FixProcessor {
...
   }
 
+  void _addFix_removeAwait() {
+    AwaitExpression awaitExpression = node;
+    if (awaitExpression != null) {
+      final awaitToken = awaitExpression.awaitKeyword;
+      _addRemoveEdit(
+          rf.rangeStartEnd(awaitToken.offset, awaitToken.next.offset));
+      _addFix(DartFixKind.REMOVE_AWAIT, []);
+    }
```
Comment (bwilkerson) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: This works, but has the disadvantage that if the node is ever not an await expression we'll get a cast exception at runtime. That's why I was recommending
```
if (node is AwaitExpression) {
  final awaitToken = node.awaitKeyword;
  ...
}
```
Worst case with the code above is that we silently don't produce a fix.

Comment (JPaulsen) [inline pkg/analysis_server/lib/src/services/correction/fix_internal.dart]: Fixed

**Review by bwilkerson — APPROVED** [review]

**Comment by alexeieleusis** [comment]
Same thing here, I am merging @bwilkerson 
We can revert if needed. Will wait a couple of minutes.

**Pull request merged**